### ISSUE TITLE
Problems with UIKit's extensions' definition of the Error enum

### ISSUE THREAD

**Issue by george-gw** [mxcl/PromiseKit#553]
* PromiseKit version: 4
* Xcode version: Version 8.0 (8A218a)
* CocoaPods version: 1.1.0.rc.2
* Swift version: 3.0

PromiseKit's extension of UIViewController caused some ambiguous errors for me. Here's my scenario:

1. My project uses GooglePlaces
2. My project uses a Pod that uses PromiseKit

```
// ViewController.swift
import UIKit
import GooglePlaces
import TheLibraryThatUsesPromiseKit

class ViewController: UIViewController {}
extension ViewController: TheGooglePlacesDelegate {}
```
GooglePlaces has a delegate that has the following function:
`func viewController(_ viewController: GMSAutocompleteViewController, didFailAutocompleteWithError error: Error)`

When my ViewController was implementing that delegate, Xcode was throwing an error saying that my ViewController isn't conforming to the delegate because the above function wasn't implemented.

It took some digging to find out that `Error` in my implementation was referring to `UIViewController`'s extension of the `enum Error` in PromiseKit rather than `Swift.Error` which is the intended Error.

So to solve this problem I had to write:
`func viewController(_ viewController: GMSAutocompleteViewController, didFailAutocompleteWithError error: Swift.Error)`
for the delegate implementation.

This wasn't straightforward to find, for two main reasons:

- Xcode's error message simply stated that the delegate doesn't conform to the protocol, and when it was trying to be helpful, it pasted the exact same code with `error: Error` rather than `error: Swift.Error`
-  I did not specifically choose to use PromiseKit

**Comment by mxcl** [comment]
Yes, good point. This was not deliberately vicious on our part since it has only occurred because Swift’s error type was renamed from `ErrorType` to `Error` during 2 -> 3 causing this conflict.

I will rename our Error types throughout.

**Comment by mxcl** [comment]
Should be fixed in 4.0.3. Please advise.

**Comment by george-gw** [comment]
It works now, thank you for fixing it so quickly :)
